### PR TITLE
Drop dependency on MPL (for TypeTraits) and Test (for LightweightTest)

### PR DESCRIPTION
**Pull request by glenfe**

### CODE REVIEW

File: test/crc_test.cpp [hunk 1]
```diff
@@ -120,9 +120,9 @@ PRIVATE_TESTER_NAME::compute_test
 
     fast_crc.process_bytes( std_data, std_data_len );
     slow_crc.process_bytes( std_data, std_data_len );
-    BOOST_CHECK( fast_crc.checksum() == expected );
-    BOOST_CHECK( slow_crc.checksum() == expected );
-    BOOST_CHECK( func_result == expected );
+    BOOST_TEST( fast_crc.checksum() == expected );
```
Comment (pdimov) [inline test/crc_test.cpp]: BOOST_TEST_EQ here?

File: test/crc_test2.cpp [hunk 2]
```diff
@@ -54,7 +48,7 @@ boost::uint16_t const  std_crc_16_result = 0xBB3Du;
 boost::uint32_t const  std_crc_32_result = 0xCBF43926ul;
 
 // Conversion functions between native- and big-endian representations
-#ifdef BOOST_BIG_ENDIAN
+#ifdef BOOST_BIG_ENDIAN_BYTE
```
Comment (pdimov) [inline test/crc_test2.cpp]: #if here. BOOST_*_ENDIAN_BYTE are always defined.

Comment (pdimov) [reply test/crc_test2.cpp]: Also, it's BOOST_ENDIAN_BIG_BYTE. :-)

Comment (glenfe) [reply test/crc_test2.cpp]: Ah. :)

File: test/crc_test.cpp (hunk 1 above)
Comment (glenfe) [reply test/crc_test.cpp]: Done.

**Comment by pdimov** [comment]
Looks good.

**Pull request merged**